spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -196,6 +204,24 @@ public void setMaxAckExtensionPeriod(Duration maxAckExtensionPeriod) {
     this.maxAckExtensionPeriod = maxAckExtensionPeriod;
   }
 
+  /**
+   * Set lower bound for a single mod ack extension period.
+   *
+   * @param minDurationPerAckExtension the lower bound for a single mod ack extension period to set
+   */
+  public void setMinDurationPerAckExtension(Duration minDurationPerAckExtension) {
+    this.minDurationPerAckExtension = minDurationPerAckExtension;
+  }
+
+  /**
+   * Set the upper bound for a single mod ack extension period.
+   *
+   * @param maxDurationPerAckExtension the upper bound for a single mod ack extension period
+   */
+  public void setMaxDurationPerAckExtension(Duration maxDurationPerAckExtension) {
+    this.maxDurationPerAckExtension = maxDurationPerAckExtension;
+  }
+
```

Comment:
Ah I wonder if we need to add new setters for new properties. They are not called anywhere else other than in tests (specifically for standalone properties like maxAckExtensionPeriod). Previously, these setters were needed because we used to directly set property valies in GcpAutoConfiguration. When we added support for per-subscription settings,  we introduced the `PubSubConfiguration` class which took care of all the logic of computing property values based on whether global (across all subscribers) or subscription-specific settings as set and modified the `DefaultSubscriberFactory` class constructor to accept a `PubSubConfiguration` object (`DefaultSubscriberFactory(
      GcpProjectIdProvider projectIdProvider, PubSubConfiguration pubSubConfiguration)`) . We could've removed these setters (including `setMaxAckExtensionPeriod` ) but decided to keep them in to avoid breaking behavior for users since they are public methods: https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/562 and https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/601 for more context.

cc @elefeint any thoughts on this?

Reply:
I think setters are mandatory actually. From [boot docs](https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.typesafe-configuration-properties.java-bean-binding):
> getters and setters are usually mandatory, since binding is through standard Java Beans property descriptors,

I probably confused the matter last year when I was looking at how property injection was done and found that for collection, a getter was used instead of the setter. The docs call collections out as an exception in which setters may be mitted.

Reply:
No, sorry, I did not notice we were in the factory. The less mutable the factory is, the better, so you are right.

Reply:
Thanks for the thorough explanation. I'll remove the setter and use `(DefaultSubscriberFactory( GcpProjectIdProvider projectIdProvider, PubSubConfiguration pubSubConfiguration)` to test user-defined value.

Reply:
done.